e2e_tests/provider_cfg/all/config.toml

```diff
@@ -18,31 +18,31 @@ admins = [ { name = "list_clients test" }, { name = "1000" }, { name = "client1"
 #workload_endpoint="unix:///tmp/agent.sock"
 
 [[key_manager]]
-name = "on-disk-manager"
-manager_type = "OnDisk"
-store_path = "./mappings"
+name = "sqlite-manager"
+manager_type = "SQLite"
+database_path = "./kim-mappings/sqlite/sqlite-key-info-manager.sqlite3"
 
 [[provider]]
 provider_type = "MbedCrypto"
-key_info_manager = "on-disk-manager"
+key_info_manager = "sqlite-manager"
```

Comment:
Matt's point here makes me think that maybe our rules around config file stability need some refinement, because I would agree that there are cases where explicitly opting in to a new feature (and hence editing the config file anyway) might prompt the need for other required changes. The current rule of "new options should be optional" is perhaps rather too vague. The real intention here is simply to ensure that existing config files always continue working, but that's not the same as saying that it should be possible to edit one thing without changing anything else.

But then there is the question about what should be done in this specific case - should a switch to the SQL lite provider prompt the addition of names for existing providers? I'm in two minds about this. It might be enough simply to ensure that our out-of-box example config includes provider names. At the moment, a working deployment cannot be "switched" from on-disk to sql anyway, because there is no migration facility being offered as yet. If and when we decide to offer a migration path, we could possibly make it a requirement of that process to inject provider names into the config.

Reply:
I have moved this conversation over to #571 !